packages/@webex/plugin-cc/src/cc.ts

```diff
@@ -1,33 +1,123 @@
+/* eslint-disable no-console */
 import {WebexPlugin} from '@webex/webex-core';
-import {CCPluginConfig, IContactCenter, WebexSDK} from './types';
-import {WCC_API_GATEWAY} from './constants';
+import {CCPluginConfig, IContactCenter, WebexSDK, CC_EVENTS, WebSocketEvent} from './types';
+import {EVENT, READY, WEBSOCKET_EVENT_TIMEOUT, SUBSCRIBE_API, WCC_API_GATEWAY} from './constants';
+import IWebSocket from './WebSocket/IWebSocket';
+import WebSocket from './WebSocket/WebSocket';
+
+const REGISTER_EVENT = 'register';
 
 export default class ContactCenter extends WebexPlugin implements IContactCenter {
   namespace = 'ContactCenter';
   $config: CCPluginConfig;
   $webex: WebexSDK;
   wccApiUrl: string;
+  webSocket: IWebSocket;
+  ciUserId: string;
+  registered = false;
+  eventHandlers: Map<
+    string,
+    {resolve: (data: any) => void; reject: (error: any) => void; timeoutId: NodeJS.Timeout}
+  > = new Map();
 
   constructor(...args) {
     super(...args);
     // @ts-ignore
     this.$config = this.config;
     // @ts-ignore
     this.$webex = this.webex;
+
+    this.$webex.once(READY, () => {
+      this.webSocket = new WebSocket({
+        parent: this.$webex,
+      });
+    });
   }
 
-  register(success: boolean): Promise<string> {
-    this.wccApiUrl = this.$webex.internal.services.get(WCC_API_GATEWAY); // Added this change for Ravi's PR, he will move this under different function if needed.
-    // TODO: Mercury Subsciption code should be added as part of this function
+  /**
+   * This is used for making the CC SDK ready by setting up the cc mercury connection.
+   * @param success
+   */
+  public register(): Promise<string> {
+    this.wccApiUrl = this.$webex.internal.services.get(WCC_API_GATEWAY);
+
+    this.listenForWebSocketEvents();
 
     return new Promise((resolve, reject) => {
-      setTimeout(() => {
-        if (success) {
-          resolve('Success: Dummy data returned');
-        } else {
-          reject(new Error('Simulated Error'));
-        }
-      }, 1000);
+      this.addEventHandler(
+        REGISTER_EVENT,
+        (result) => {
+          this.registered = true;
+          resolve(result);
+        },
+        reject
+      );
+
+      this.establishConnection();
+    });
+  }
+
+  /**
+   * This is used for unregistering the CC SDK by disconnecting the cc mercury connection.
+   * @returns Promise<void>
+   */
+  public unregister(): Promise<void> {
+    return this.webSocket.disconnectWebSocket().then(() => {
+      this.webSocket.off(EVENT, this.processEvent);
```

Comment:
Would it make sense to switch off the listener after the disconnect has already happened? This should happened in disconnectWebsocket function itself I believe

Reply:
Can we get a response on this also?